### PR TITLE
[DOCS] Split off ILM overview to a separate topic.

### PR DESCRIPTION
**Pull request by debadair**

### CODE REVIEW

File: docs/reference/ilm/index.asciidoc [hunk 1]
```diff
@@ -1,72 +1,29 @@
...
-
-[IMPORTANT]
-===========================
-{ilm} does not support mixed-version cluster usage. Although it
-may be possible to create such new policies against
-newer-versioned nodes, there is no guarantee they will
-work as intended. New policies using new actions that
-do not exist in the oldest versioned node will cause errors.
-===========================
+You can configure index lifecycle policies to automatically manage indices as they age.
```
Comment (jrodewig) [inline docs/reference/ilm/index.asciidoc]:
```suggestion
You can configure index lifecycle policies to automatically manage indices as they age or grow. 
```
This sort of implies that you can only manage indices based on dates.

File: docs/reference/ilm/index.asciidoc [hunk 2]
```diff
@@ -1,72 +1,29 @@
...
-===========================
-{ilm} does not support mixed-version cluster usage. Although it
-may be possible to create such new policies against
-newer-versioned nodes, there is no guarantee they will
-work as intended. New policies using new actions that
-do not exist in the oldest versioned node will cause errors.
-===========================
+You can configure index lifecycle policies to automatically manage indices as they age. 
+Index lifecycle policies can trigger actions such as rollover, shrink, force merge, freeze, and delete 
+based on your performance, resiliency, and retention requirements.
```
Comment (jrodewig) [inline docs/reference/ilm/index.asciidoc]: I don't know if we should assume users are familiar with actions like "shrink" or "rollover." It help to reword this as a short bulleted list of examples, focusing on the biggest use cases.

>You can use index lifecycle policies to automatically:
> * Create a new index each day, week, or month and archive previous ones
> * Spin up a new index when an index reaches a certain size
> * Delete stale indices to enforce data retention standards

File: docs/reference/ilm/index.asciidoc [hunk 3]
```diff
@@ -1,72 +1,29 @@
...
-may be possible to create such new policies against
-newer-versioned nodes, there is no guarantee they will
-work as intended. New policies using new actions that
-do not exist in the oldest versioned node will cause errors.
-===========================
+You can configure index lifecycle policies to automatically manage indices as they age. 
+Index lifecycle policies can trigger actions such as rollover, shrink, force merge, freeze, and delete 
+based on your performance, resiliency, and retention requirements. 
+To automatically back up your indices and manage snapshots, 
+use <<getting-started-snapshot-lifecycle-management,snapshot lifecycle policies>>.
```
Comment (jrodewig) [inline docs/reference/ilm/index.asciidoc]: I'd put this in a `[TIP]` block as it's not directly related.

File: docs/reference/ilm/index.asciidoc [hunk 4]
```diff
@@ -1,72 +1,29 @@
...
+
+* <<overview-index-lifecycle-management, Overview>>
+* <<getting-started-index-lifecycle-management, Getting Started with {ilm-init}>>
+* <<ilm-policy-definition, Policy phases and actions>>
+* <<set-up-lifecycle-policy, Creating policies>>
```
Comment (jrodewig) [inline docs/reference/ilm/index.asciidoc]: I'm not a huge fan of gerund titles. Not a huge deal though.

File: docs/reference/ilm/overview-ilm.asciidoc [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+[[overview-index-lifecycle-management]]
+== Index lifecycle management overview
+
+++++
+<titleabbrev>Overview</titleabbrev>
+++++
+
+You can create and apply {ilm-cap} ({ilm-init}) policies to automatically manage your indices 
+according to your performance, resiliency, and retention requirements. 
+Index lifecycle policies can trigger actions such as rollover, shrink, freeze, and delete.
```
Comment (jrodewig) [inline docs/reference/ilm/overview-ilm.asciidoc]: I'm not sure a new user would know what these actions are or what they mean.
It may be more helpful to do one of the following:

* Remove this sentence
* Add links to the example actions
* Reword this to go into a bit more depth

File: docs/reference/ilm/overview-ilm.asciidoc [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+++++
+<titleabbrev>Overview</titleabbrev>
+++++
+
+You can create and apply {ilm-cap} ({ilm-init}) policies to automatically manage your indices 
+according to your performance, resiliency, and retention requirements. 
+Index lifecycle policies can trigger actions such as rollover, shrink, freeze, and delete. 
+
+Typically, you associate a lifecycle policy with an index template so it is applied   
+whenever an index is created from the template.
```
Comment (jrodewig) [inline docs/reference/ilm/overview-ilm.asciidoc]: I thought the way it worked was:

* You define a match pattern when you create an index template.
* When you create an index that matches the match pattern, the index template is applied.

This sort of implies you create indices from the template space, which I don't think is the case.

File: docs/reference/ilm/overview-ilm.asciidoc [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+You can create and apply {ilm-cap} ({ilm-init}) policies to automatically manage your indices 
+according to your performance, resiliency, and retention requirements. 
+Index lifecycle policies can trigger actions such as rollover, shrink, freeze, and delete. 
+
+Typically, you associate a lifecycle policy with an index template so it is applied   
+whenever an index is created from the template. 
+You can also apply a policy manually when you create an index.
+
+Index lifecycle policies are most often used to manage time-series indices. 
+Time-series indices have a four stage lifecycle:
```
Comment (jrodewig) [inline docs/reference/ilm/overview-ilm.asciidoc]: I think I'd frame this as "ILM is often used to manage hot-cold architecture. Hot-warm-cold architectures are common for time series data, such as logging or metrics."

In any case, the second sentence implies that time-series data always has a hot-cold architecture, which isn't necessarily true. I think it also confuses hot-cold architecture with time-series data. Hot-cold architecture is helpful in contexts outside of time-series data.

I'd also note that hot-cold architecture _can_ include up to four stages. For example, you could easily pull out the `warm` or `delete` stages.

File: docs/reference/ilm/overview-ilm.asciidoc [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+* Hot--the index is actively being updated and queried.
+* Warm--the index is no longer being updated, but is still being queried.
+* Cold--the index is no longer being updated and is seldom queried. The
+information still needs to be searchable, but it's okay if those queries are
+slower.
+* Delete--the index is no longer needed and can safely be deleted.
+
+A lifecycle policy governs how the index transitions through these stages and
```
Comment (jrodewig) [inline docs/reference/ilm/overview-ilm.asciidoc]:
```suggestion
You can use a lifecycle policy to govern how an index transitions through these stages and
```

File: docs/reference/ilm/overview-ilm.asciidoc [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+the actions that are performed on the index at each stage.
```
Comment (jrodewig) [inline docs/reference/ilm/overview-ilm.asciidoc]:
```suggestion
the actions performed at each stage. 
```

File: docs/reference/ilm/overview-ilm.asciidoc [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+. When the index reaches 50GB, roll over to a new index.
+. Move the old index into the warm stage, mark it read only, and shrink it down
+to a single shard.
+. After 7 days, move the index into the cold stage and move it to less expensive
+hardware.
+. Delete the index once the required 30 day retention period is reached.
+
+[IMPORTANT]
+===========================
+{ilm} does not support mixed-version clusters.
```
Comment (jrodewig) [inline docs/reference/ilm/overview-ilm.asciidoc]: We should probably explain what a mixed-version cluster is.

**Review by jrodewig — APPROVED** [review]
I left a few comments where I think we could improve, but nothing I would consider blocking. Thanks for putting this together @debadair.

File: docs/reference/ilm/overview-ilm.asciidoc (hunk 8 above)
Comment (debadair) [reply docs/reference/ilm/overview-ilm.asciidoc]: I tweaked this a bit, but left the focus on the policy driving the transitions and actions.
